Library/ViewModels/PledgeViewModel.swift

```diff
@@ -156,40 +156,17 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       rewards,
       selectedQuantities
     )
-    .map { rewards, selectedQuantities -> Double in
-      rewards.filter { !$0.isNoReward }
-        .reduce(0.0) { total, reward -> Double in
-          let totalForReward = reward.minimum
-            .multiplyingCurrency(Double(selectedQuantities[reward.id] ?? 0))
-
-          return total.addingCurrency(totalForReward)
-        }
-    }
+    .map(calculateAllRewardsTotal)
 
     let calculatedShippingTotal = Signal.combineLatest(
       selectedShippingRule.skipNil(),
       rewards,
       selectedQuantities
     )
-    .map { shippingRule, rewards, selectedQuantities -> Double in
-      rewards.reduce(0.0) { total, reward in
-        guard reward.shipping.enabled else { return total }
-
-        let shippingCostForReward = reward.shippingRule(matching: shippingRule)?.cost ?? 0
-
-        let totalShippingForReward = shippingCostForReward
-          .multiplyingCurrency(Double(selectedQuantities[reward.id] ?? 0))
-
-        return total.addingCurrency(totalShippingForReward)
-      }
-    }
-
-    let defaultShippingTotal = Signal.zip(project, baseReward)
-      .map { project, baseReward -> Double in
-        guard baseReward.shipping.enabled, let backing = project.personalization.backing else { return 0.0 }
+    .map(calculateShippingTotal)
 
-        return backing.shippingAmount.flatMap(Double.init) ?? 0.0
-      }
+    let defaultShippingTotal = Signal.zip(project, baseReward, selectedShippingRule)
```

Comment:
a better name would be related to the method being applied. something like `baseRewardShippingTotal`